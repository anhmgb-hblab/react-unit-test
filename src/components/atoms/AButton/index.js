import React from 'react';
import { Button } from 'antd';
import PropTypes from 'prop-types';

const AButton = ({
  type,
  children,
  size,
  onClick,
  disabled,
  loading,
  icon,
  ...props
}) => (
  <Button
    type={type}
    size={size}
    onClick={onClick}
    loading={loading}
    disabled={disabled}
    icon={icon}
    {...props}
  >
    {children}
  </Button>
);

AButton.defaultProps = {
  type: 'primary',
  size: 'middle',
  onClick: null,
  loading: false,
  disabled: false,
  icon: null,
  children: null,
};

AButton.propTypes = {
  type: PropTypes.string,
  size: PropTypes.string,
  children: PropTypes.node,
  onClick: PropTypes.func,
  loading: PropTypes.bool,
  disabled: PropTypes.bool,
  icon: PropTypes.node,
};

export default AButton;
