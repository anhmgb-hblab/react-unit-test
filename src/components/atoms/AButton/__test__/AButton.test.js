import React from 'react';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import AButton from '../index';

describe('Test AButton running correctly', () => {
  it('test button props', () => {
    const wrapper = mount(<AButton>Primary</AButton>);
    expect(wrapper.props().size).toEqual('middle');
    expect(wrapper.props().type).toEqual('primary');
    expect(wrapper.props().loading).toEqual(false);
    expect(wrapper.props().disabled).toEqual(false);
    expect(wrapper.text()).toEqual('Primary');
  });

  it('test button onClick', () => {
    const onClick = jest.fn(() => 'clicked');
    const wrapper = mount(<AButton onClick={onClick}>Primary</AButton>);
    wrapper.props().onClick();

    expect(onClick).toHaveBeenCalledTimes(1);
    expect(onClick.mock.results[0].value).toBe('clicked');
  });

  it('test rendering', () => {
    const tree = mount(<AButton>Primary</AButton>);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
