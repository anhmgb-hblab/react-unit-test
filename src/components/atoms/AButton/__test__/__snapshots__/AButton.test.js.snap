// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AButton running correctly test rendering 1`] = `
<AButton
  disabled={false}
  icon={null}
  loading={false}
  onClick={null}
  size="middle"
  type="primary"
>
  <Button
    disabled={false}
    icon={null}
    loading={false}
    onClick={null}
    size="middle"
    type="primary"
  >
    <Wave>
      <button
        className="ant-btn ant-btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <LoadingIcon
          existIcon={false}
          loading={false}
          prefixCls="ant-btn"
        >
          <CSSMotion
            motionName="ant-btn-loading-icon-motion"
            onAppearActive={[Function]}
            onAppearStart={[Function]}
            onEnterActive={[Function]}
            onEnterStart={[Function]}
            onLeaveActive={[Function]}
            onLeaveStart={[Function]}
            removeOnLeave={true}
            visible={false}
          >
            <DomWrapper />
          </CSSMotion>
        </LoadingIcon>
        <span
          key=".0"
        >
          Primary
        </span>
      </button>
    </Wave>
  </Button>
</AButton>
`;
